include/d/actor/d_a_tsubo.h

```diff
@@ -13,28 +13,166 @@
+            /* 0x72 */ u8 mAttnDist;
+            /* 0x74 */ Vec m74;
+            /* 0x80 */ u32 m80;
+            /* 0x84 */ s32 mSoundID_Break;
+            /* 0x88 */ s32 m88;
+            /* 0x8C */ s32 mSoundID_FallLava;
+            /* 0x90 */ s32 mSoundID_FallWater;
+            /* 0x94 */ s32 mSoundID_Hit;
+            /* 0x98 */ s16 mCullSphX_Move;
+            /* 0x9A */ s16 mCullSphY_Move;
+            /* 0x9C */ s16 mCullSphZ_Move;
+            /* 0x9E */ s16 mCullSphR_Move;
+            /* 0xA0 */ u8 mA0[0xA4 - 0xA0];
+            /* 0xA4 */ s16 mCullSphX_Draw;
+            /* 0xA6 */ s16 mCullSphY_Draw;
+            /* 0xA8 */ s16 mCullSphZ_Draw;
+            /* 0xAA */ s16 mCullSphR_Draw;
+            /* 0xAC */ u8 mAC[0xB0 - 0xAC];
+            /* 0xB0 */ f32 mB0;
+            /* 0xB4 */ u8 mB4[0xB8 - 0xB4];
+            /* 0xB8 */ u32 mHeapSize;
+            /* 0xBC */ f32 mBC;
+            /* 0xC0 */ f32 mC0;
+            /* 0xC4 */ f32 mC4;
+            /* 0xC8 */ f32 mC8;
+        }; // size = 0xCC
+
+        struct AttrSpine_c {
+            /* 0x00 */ f32 m00;
+            /* 0x04 */ f32 m04;
+            /* 0x08 */ f32 m08;
+            /* 0x0C */ f32 m0C;
+            /* 0x10 */ f32 m10;
+            /* 0x14 */ f32 m14;
+            /* 0x18 */ s16 m18;
+            /* 0x1A */ s16 m1A;
+            /* 0x1C */ s16 m1C;
+            /* 0x1E */ s16 m1E;
+            /* 0x20 */ f32 m20;
+            /* 0x24 */ f32 m24;
+            /* 0x28 */ f32 m28;
+            /* 0x2C */ f32 m2C;
+            /* 0x30 */ s16 m30;
+        }; // size = 0x34
+
+        struct SpecBokoData {
+            /* 0x00 */ s16 m00;
+            /* 0x02 */ s16 m02;
+            /* 0x04 */ f32 m04;
+            /* 0x08 */ f32 m08;
+        }; // size = 0xC
+
+        struct M7A0 {
+            /* 0x00 */ fpc_ProcID m00;
+            /* 0x04 */ u8 m04;
+            /* 0x06 */ s16 m06;
+            /* 0x08 */ s16 m08;
+            /* 0x0C */ f32 m0C;
+            /* 0x10 */ f32 m10;
+            /* 0x14 */ f32 m14;
+            /* 0x18 */ s8 m18;
+        }; // size = 0x1C
         
         int prm_get_type() const { return daObj::PrmAbstract(this, PRM_TYPE_W, PRM_TYPE_S); }
         bool prm_chk_type_kutani() const { return prm_get_type() == TYPE_KUTANI; }
         
-        void attrSpine() {}
-        void data() const {}
-        void data_spec_boko(int) {}
-        void is_switch() const {}
+        const AttrSpine_c& attrSpine() { return M_attrSpine; }
+        const Data_c& data() const { return M_data[mType]; }
+        const SpecBokoData& data_spec_boko(int index) { return M_data_spec_boko[index]; }
+        #if VERSION == VERSION_DEMO
+        bool is_switch() const { 
+            return fopAcM_isSwitch(const_cast<Act_c*>(this), prmZ_get_swSave());
+        }
+        #else
+        BOOL is_switch() const { 
+            int sw = prmZ_get_swSave();
+            return fopAcM_isSwitch(const_cast<Act_c*>(this), sw);
+        }
+        #endif
```

Comment:
These two are equivalent so it's unlikely it changed in between versions. Let's avoid using version ifdefs for fakematches as it just obscures the real match (e.g. we might still need to change the compiler flags for demo to fix certain things, the current ones are just my guess and are very likely incomplete). Someone else can come back later and find the real fix that matches both versions at once, but for now it's fine for it to only match on retail.
```suggestion
        BOOL is_switch() const { 
            int sw = prmZ_get_swSave();
            return fopAcM_isSwitch(const_cast<Act_c*>(this), sw);
        }
```